ios/Integreat/AppDelegate.m

```diff
@@ -46,4 +48,39 @@ - (NSURL *)sourceURLForBridge:(RCTBridge *)bridge
 #endif
 }
 
+- (void)clearStorage
+{
+  NSUserDefaults *preferences = [NSUserDefaults standardUserDefaults];
+  NSString *lastLanguageKey = @"last_location";
+
+  if ([preferences objectForKey:lastLanguageKey] != nil)
+  {
+    //clear preferences
+    [self resetDefaults];
+    
+    //clear storage
+    [self removeCache];
+  }
+}
+
+- (void)resetDefaults {
+    NSUserDefaults * defs = [NSUserDefaults standardUserDefaults];
+    NSDictionary * dict = [defs dictionaryRepresentation];
+    for (id key in dict) {
+        [defs removeObjectForKey:key];
+    }
+    [defs synchronize];
+}
+
+- (void)removeCache
+{
+    NSFileManager* manager = [NSFileManager defaultManager];
+    NSArray *paths = NSSearchPathForDirectoriesInDomains(NSLibraryDirectory, NSUserDomainMask, YES);
+    NSString *libraryDirectory = [paths objectAtIndex:0];
+    NSArray *directoryContent = [manager directoryContentsAtPath: libraryDirectory];
+    for (NSString *content in directoryContent)  {
```

Comment:
Not sure whether we need to delete stuff recursively here.

The thing is that we need to be 100% sure that the state of the app is like if the RN app was installed freshly.